spark_client/services.py

```diff
@@ -112,6 +96,482 @@ def user(self):
+
+        Args:
+            resource_type: type of the resource to be labeled, e.g. service account, rolebindings, etc.
+            resource_name: name of the resource to be labeled
+            label: label to remove
+            namespace: namespace where the resource is
+        """
+        label_to_remove = f"/metadata/labels/{label.replace('/','~1')}"
+        self.logger.debug(f"Removing label {label_to_remove}")
+        patch = [{"op": "remove", "path": label_to_remove}]
+
+        if resource_type == "serviceaccount":
+            self.client.patch(
+                res=LightKubeServiceAccount,
+                name=resource_name,
+                namespace=namespace,
+                obj=patch,
+                patch_type=PatchType.JSON,
+            )
+        elif resource_type == "role":
+            self.client.patch(
+                res=Role,
+                name=resource_name,
+                namespace=namespace,
+                obj=patch,
+                patch_type=PatchType.JSON,
+            )
+        elif resource_type == "rolebinding":
+            self.client.patch(
+                res=RoleBinding,
+                name=resource_name,
+                namespace=namespace,
+                obj=patch,
+                patch_type=PatchType.JSON,
+            )
+        else:
+            raise NotImplementedError(
+                f"Label setting for resource name {resource_type} not supported yet."
+            )
+
+    def create_property_file_entries(self, property_file_name) -> Dict[str, str]:
+        entries = dict()
+        props = PropertyFile.read(property_file_name).props
+        for k in props:
+            entries[k] = base64.b64encode(str(props[k]).encode("ascii"))
+        return props
+
+    def create(
+        self,
+        resource_type: str,
+        resource_name: str,
+        username: str,
+        namespace: str,
+        **extra_args,
+    ):
+        """Create a K8s resource.
+
+        Args:
+            resource_type: type of the resource to be created, e.g. service account, rolebindings, etc.
+            resource_name: name of the resource to be created
+            namespace: namespace where the resource is
+            extra_args: extra parameters that should be provided when creating the resource. Note that each parameter
+                        will be prepended with the -- in the cmd, e.g. {"role": "view"} will translate as
+                        --role=view in the command. List of parameter values against a parameter key are also accepted.
+                        e.g. {"resource" : ["pods", "configmaps"]} which would translate to something like
+                        --resource=pods --resource=configmaps
+        """
+
+        res = None
+        if resource_type == "serviceaccount":
+            with open(
+                f"{os.getenv('SNAP')}/python/spark_client/resources/lightkube_jinja_templates/serviceaccount_yaml.tmpl"
```

Comment:
**issue** this won't work outside of snap. Beside, in general we should never access envs using os.getenv. We defined the Defaults object to be able to proxy and centralize the envs that we use, such that we have control on these. If we use os.env scattered around, we then lose this control 

Anyhow, these are resources and as such, there is a pythonic way to access them which is using this library

https://importlib-resources.readthedocs.io/en/latest/using.html